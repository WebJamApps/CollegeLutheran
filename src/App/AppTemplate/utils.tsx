function showNav(isAllowed: boolean, location: { pathname: string; }, m: { link: string; }, auth: { isAuthenticated: boolean; }) {
  if (!isAllowed) return false;
  if (location.pathname === '/staff' && m.link === '/staff') return false;
  if ((m.link === '/staff' || m.link === '/belief') && auth.isAuthenticated) return false;
  return true;
}

export default { showNav };
