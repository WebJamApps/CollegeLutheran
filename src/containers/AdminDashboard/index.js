import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import mapStoreToProps from '../../redux/mapStoreToProps';
import forms from '../../lib/forms';
import AdminController from './AdminController';
import commonUtils from '../../lib/commonUtils';

export class AdminDashboard extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
    this.controller = new AdminController(this);
    this.state = {
      title: '',
      homePageContent: '',
      forumtitle: '',
      forumurl: '',
      youthName: '',
      youthURL: '',
      forumId: '',
      youthPicsId: '',
      childName: '',
      childURL: '',
      familyPicsId: '',
    };
    this.forms = forms;
    this.onChange = this.onChange.bind(this);
    this.changeHomepage = this.changeHomepage.bind(this);
    this.addForumForm = this.addForumForm.bind(this);
    this.changePicForm = this.changePicForm.bind(this);
    this.deleteFamily = this.deleteFamily.bind(this);
    this.changeFamilyForm = this.changeFamilyForm.bind(this);
    this.deleteForumForm = this.deleteForumForm.bind(this);
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Admin Dashboard'); }

  onChange(evt, stateValue) {
    return typeof stateValue === 'string' ? this.setState({ [stateValue]: evt.target.value })
      : this.setState({ [evt.target.id]: evt.target.value });
  }

  deleteFamily() {
    const { familyPicsId } = this.state;
    const { familyPics } = this.props;
    return (
      <form
        id="delete-family"
        style={{
          textAlign: 'left', marginLeft: '4px', width: '100%', maxWidth: '100%',
        }}
      >
        { this.forms.makeDropdown('familyPicsId', '* Select Pic to Delete', familyPicsId, this.onChange, familyPics, '_id', 'title') }
        <div style={{ marginLeft: '60%' }}>
          <p>{' '}</p>
          <button
            onClick={(evt) => this.controller.deleteBookApi(evt, familyPicsId, '/family')}
            type="button"
            disabled={this.controller.validateDeleteBook(familyPicsId)}
          >
            Delete Pic
          </button>
        </div>
      </form>
    );
  }

  changePicForm(picData) {
    const imageUrlValue = this.state[picData.urlId];// eslint-disable-line react/destructuring-assignment
    const imageNameValue = this.state[picData.nameId];// eslint-disable-line react/destructuring-assignment
    return (
      <div className="material-content elevation3" style={{ maxWidth: '320px', margin: 'auto' }}>
        <h4 className="material-header-h4">
          Change
          {' '}
          {picData.title}
          {' '}
          Pictures
        </h4>
        <form>
          <label htmlFor={picData.nameId}>
            Picture Name
            <input id={picData.nameId} value={imageNameValue} onChange={this.onChange} />
          </label>
          <label htmlFor={picData.urlId}>
            Image Address
            <input id={picData.urlId} value={imageUrlValue} onChange={this.onChange} />
          </label>
          <div style={{ marginLeft: '70%' }}>
            <p>{' '}</p>
            <button disabled={picData.disabled()} type="button" id={picData.buttonId} onClick={picData.buttonClick}>Add Pic</button>
          </div>
        </form>
        <hr />
        {picData.deleteSection()}
      </div>
    );
  }

  deleteForumForm(forumId, books) {
    return (
      <form
        id="delete-forum"
        style={{
          textAlign: 'left', margin: 'auto', width: '100%', maxWidth: '100%',
        }}
      >
        { this.forms.makeDropdown('forumId', '* Select Forum to Delete', forumId, this.onChange, books, '_id', 'title') }
        <p>{' '}</p>
        <button
          onClick={(evt) => this.controller.deleteBookApi(evt, forumId, '/news')}
          type="button"
          disabled={this.controller.validateDeleteBook(forumId)}
        >
          Delete Forum
        </button>
      </form>
    );
  }

  addForumForm() {
    const { forumtitle, forumurl, forumId } = this.state;
    const { books } = this.props;
    return (
      <div className="material-content elevation3" style={{ maxWidth: '8in', margin: 'auto' }}>
        <h5>Change Monthly Forum</h5>
        <form
          id="create-forum"
          style={{
            textAlign: 'left', marginLeft: '4px', width: '100%', maxWidth: '100%',
          }}
        >
          {this.forms.makeInput('text', 'Forum Title', false, this.onChange, forumtitle, '90%')}
          {this.forms.makeInput('text', 'Forum URL', false, this.onChange, forumurl, '90%')}
          <div style={{ marginLeft: '70%' }}>
            <p>{' '}</p>
            <button
              type="button"
              id="addForum"
              disabled={this.controller.validateBook(forumtitle, forumurl)}
              onClick={this.controller.addForumAPI}
            >
              Add Forum
            </button>
          </div>
        </form>
        <hr />
        {this.deleteForumForm(forumId, books)}
      </div>
    );
  }

  changeHomepage() {
    const { title, homePageContent } = this.state;
    return (
      <div className="material-content elevation3" style={{ maxWidth: '8in', margin: 'auto' }}>
        <h5>Change Homepage Section</h5>
        <form
          id="create-homepage"
          style={{
            textAlign: 'left', marginLeft: '4px', width: '100%', maxWidth: '100%',
          }}
        >
          {this.forms.makeInput('text', 'Title', false, this.onChange, title, '90%')}
          <label htmlFor="content">
            Content
            <br />
            <textarea id="homePageContent" rows="15" value={homePageContent} style={{ width: '90%' }} onChange={this.onChange} />
          </label>
          <div style={{ marginLeft: '60%' }}>
            <button type="button" id="c-h" disabled={false} onClick={this.controller.createHomeAPI}>Update Homepage</button>
          </div>
        </form>
      </div>
    );
  }

  changeFamilyForm(childName, childURL) {
    return (
      this.changePicForm({
        title: 'Family',
        nameId: 'childName',
        urlId: 'childURL',
        disabled: () => this.controller.validateBook(childName, childURL),
        buttonId: 'addFamilyPic',
        buttonClick: (e) => this.controller.createPicApi(e, {
          title: childName,
          url: childURL,
          comments: childURL,
          type: 'familyPics',
          access: 'CLC',
        }, '/family'),
        deleteSection: this.deleteFamily,
      })
    );
  }

  render() {
    const {
      youthName, youthPicsId, youthURL, childName, childURL,
    } = this.state;
    const { youthPics } = this.props;
    return (
      <div className="page-content">
        <h4 style={{ textAlign: 'center', marginTop: '10px' }}>CLC Admin Dashboard</h4>
        {this.changeHomepage()}
        <p>{' '}</p>
        {this.addForumForm()}
        <p>{' '}</p>
        {this.changePicForm({
          title: 'Youth',
          nameId: 'youthName',
          urlId: 'youthURL',
          disabled: () => this.controller.validateBook(youthName, youthURL),
          buttonId: 'addYouthPic',
          buttonClick: (e) => this.controller.createPicApi(e, {
            title: youthName,
            url: youthURL,
            comments: youthURL,
            type: 'youthPics',
            access: 'CLC',
          }, '/youth'),
          deleteSection: () => this.controller.deleteBookForm('youthPicsId', 'Pic', youthPicsId, youthPics, '/youth'),
        })}
        <p>{' '}</p>
        {this.changeFamilyForm(childName, childURL)}
      </div>
    );
  }
}
AdminDashboard.propTypes = {
  auth: PropTypes.shape({ token: PropTypes.string }).isRequired,
  books: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  youthPics: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  familyPics: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};
export default withRouter(connect(mapStoreToProps)(AdminDashboard));
