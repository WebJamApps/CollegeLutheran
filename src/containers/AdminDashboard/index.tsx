import React, { Component, Dispatch } from 'react';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';
import mapStoreToProps, { Ibook } from '../../redux/mapStoreToProps';
import forms from '../../lib/forms';
import AdminController from './AdminController';
import commonUtils from '../../lib/commonUtils';
import PTable from '../../components/PhotoTable';

export interface PicData {
  buttonId: string; buttonClick: (e: React.ChangeEvent<EventTarget>) => Promise<string>; title: string; nameId: string;
}
export interface DashboardProps extends RouteComponentProps {
  dispatch: Dispatch<unknown>;
  homeContent: Ibook;
  auth: { token: string };
  books: Record<string, string>[];
  showTable: boolean;
  editPic: Ibook;
  youthPics: Ibook[];
  familyPics: Ibook[];
  otherPics: Ibook[];
  musicPics: Ibook[];
}
type DashboardState = {
  type: string;
  title: string;
  homePageContent: string;
  announcementtitle: string;
  announcementurl: string;
  youthName: string;
  youthURL: string;
  forumId: string;
  showCaption: string;
  firstEdit: boolean;
};
export class AdminDashboard extends Component<DashboardProps, DashboardState> {
  commonUtils: { setTitleAndScroll: (pageTitle: string, width: number) => void; };

  controller: AdminController;

  forms: typeof forms;

  constructor(props: DashboardProps) {
    super(props);
    this.commonUtils = commonUtils;
    this.controller = new AdminController(this);
    this.state = {
      type: '',
      title: props.homeContent.title || '',
      homePageContent: props.homeContent.comments || '',
      announcementtitle: '',
      announcementurl: '',
      youthName: '',
      youthURL: '',
      forumId: '',
      showCaption: '',
      firstEdit: true,
    };
    this.forms = forms;
    this.onChange = this.onChange.bind(this);
    this.onChangeSelect = this.onChangeSelect.bind(this);
    this.checkEdit = this.checkEdit.bind(this);
    this.changeHomepage = this.changeHomepage.bind(this);
    this.changePicForm = this.changePicForm.bind(this);
    this.deleteForumForm = this.deleteForumForm.bind(this);
    this.picButton = this.picButton.bind(this);
    this.handleRadioChange = this.handleRadioChange.bind(this);
    this.resetEditForm = this.resetEditForm.bind(this);
  }

  componentDidMount(): void { this.commonUtils.setTitleAndScroll('Admin Dashboard', window.screen.width); }

  onChange(evt: React.ChangeEvent<HTMLInputElement>, stateValue?: string): string {
    evt.persist();
    this.checkEdit();
    if (typeof stateValue === 'string') {
      this.setState((prevState) => ({ ...prevState, [stateValue]: evt.target.value, firstEdit: false }));
      return stateValue;
    }
    this.setState((prevState) => ({ ...prevState, [evt.target.id]: evt.target.value, firstEdit: false }));
    return evt.target.id;
  }

  onChangeSelect(evt: React.ChangeEvent<HTMLSelectElement>, stateValue?: string): string {
    evt.persist();
    this.checkEdit();
    if (typeof stateValue === 'string') {
      this.setState((prevState) => ({ ...prevState, [stateValue]: evt.target.value }));
      return stateValue;
    }
    this.setState((prevState) => ({ ...prevState, [evt.target.id]: evt.target.value }));
    return evt.target.id;
  }

  checkEdit(): void {
    let {
      youthName, youthURL, type, showCaption,
    } = this.state;
    const { editPic } = this.props;
    if (youthName === '' && editPic.title !== undefined) {
      youthName = editPic.title;
    }
    if (youthURL === '' && editPic.url !== undefined) {
      youthURL = editPic.url;
    }
    if (type === '' && editPic.type !== undefined) {
      type = editPic.type;
    }
    if (showCaption === '' && editPic.comments !== undefined) {
      showCaption = editPic.comments;
    }
    this.setState({
      youthName, youthURL, type, showCaption,
    });
  }

  resetEditForm(evt: React.MouseEvent<HTMLButtonElement>): void {
    evt.preventDefault();
    const { dispatch } = this.props;
    dispatch({ type: 'EDIT_PIC', picData: {} });
    dispatch({ type: 'SHOW_TABLE', showTable: true });
    this.setState({
      youthName: '', youthURL: '', type: '', showCaption: '', firstEdit: true,
    });
  }

  picButton(picData: PicData,
    editPic: { _id: string }, youthName: string, youthURL: string, type: string): JSX.Element {
    const { firstEdit } = this.state;
    return (
      <div style={{ marginLeft: '50%', marginTop: '10px' }}>
        {editPic._id ? (
          <button
            style={{ display: 'relative', marginRight: '20px' }}
            type="button"
            id="cancel-edit-pic"
            onClick={this.resetEditForm}
          >
            Cancel
          </button>
        ) : null}
        <button
          style={{ display: 'relative' }}
          disabled={this.controller.validateBook(youthName, youthURL, type, firstEdit)}
          type="button"
          id={picData.buttonId}
          onClick={
            editPic._id ? this.controller.editPicAPI : picData.buttonClick
          }
        >
          {editPic._id ? 'Edit ' : 'Add '}
          Pic
        </button>
      </div>
    );
  }

  handleRadioChange(evt: { target: { value: string } }): void {
    this.checkEdit();
    this.setState({ showCaption: evt.target.value });
  }

  changePicForm(picData: PicData): JSX.Element['props'] {
    const options = [
      { type: 'youthPics', Category: 'Youth Pics' },
      { type: 'familyPics', Category: 'Family Pics' },
      { type: 'otherPics', Category: 'Other Pics' },
      { type: 'musicPics', Category: 'Music Pics' },
    ];
    const { youthURL, youthName } = this.state;
    let { type, showCaption } = this.state;
    const { editPic } = this.props;
    if (type === '' && editPic.type !== undefined) {
      type = editPic.type;
    }
    if (showCaption === '' && editPic.comments !== undefined) {
      showCaption = editPic.comments;
    }
    return this.controller.changePicDiv(editPic, youthName, youthURL, type, options, showCaption, picData);
  }

  deleteForumForm(forumId: string, options: Record<string, string>[]): JSX.Element {
    const ddParams = {
      htmlFor: 'forumId',
      labelText: '* Select Title to Delete',
      value: forumId,
      onChange: this.onChangeSelect,
      options,
      oValue: '_id',
      dValue: 'title',
    };
    return (
      <form
        id="delete-forum"
        style={{
          textAlign: 'left', margin: 'auto', width: '100%', maxWidth: '100%',
        }}
      >
        {this.forms.makeDataDropdown(ddParams)}
        <p> </p>
        <button
          onClick={(evt) => this.controller.deleteBookApi(evt, forumId, '/news')}
          type="button"
          disabled={this.controller.validateDeleteBook(forumId)}
        >
          Delete Announcement
        </button>
      </form>
    );
  }

  changeHomepage(): JSX.Element {
    const { title, homePageContent } = this.state;
    const inputParams = {
      type: 'text', label: 'Title', isRequired: false, onChange: this.onChange, value: title, width: '90%',
    };
    return (
      <div className="horiz-scroll">
        <div className="material-content elevation3" style={{ width: '850px', margin: '30px auto' }}>
          <h5>Change Homepage Section</h5>
          <form
            id="create-homepage"
            style={{
              textAlign: 'left', marginLeft: '4px', width: '100%', maxWidth: '100%',
            }}
          >
            {this.forms.makeInput(inputParams)}
            <label htmlFor="content">
              Content
              <br />
              {this.controller.editor(homePageContent)}
            </label>
            <div style={{ marginLeft: '60%', marginTop: '10px' }}>
              <button type="button" id="c-h" disabled={false} onClick={this.controller.createHomeAPI}>
                Update Homepage
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }

  changeYouthForm(): string {
    const {
      youthName, youthURL, type, showCaption,
    } = this.state;
    const postBody = {
      title: youthName,
      url: youthURL,
      comments: showCaption,
      type,
      access: 'CLC',
    };
    return this.changePicForm({
      buttonId: 'addYouthPic',
      buttonClick: (e: React.ChangeEvent<EventTarget>) => this.controller.createPicApi(e, postBody, '/admin'),
      title: '',
      nameId: 'youthName',
    });
  }

  render(): JSX.Element {
    const {
      showTable, auth, dispatch, youthPics, familyPics, otherPics, musicPics,
    } = this.props;
    return (
      <div className="page-content">
        <h4 style={{ textAlign: 'center', marginTop: '10px' }}>
          CLC Admin Dashboard
        </h4>
        {this.changeHomepage()}
        {this.controller.addForumForm()}
        {this.changeYouthForm()}
        {showTable ? (
          <PTable
            auth={auth}
            dispatch={dispatch}
            youthPics={youthPics}
            familyPics={familyPics}
            otherPics={otherPics}
            musicPics={musicPics}
          />
        ) : null}
      </div>
    );
  }
}

export default withRouter(connect(mapStoreToProps, null)(AdminDashboard));
