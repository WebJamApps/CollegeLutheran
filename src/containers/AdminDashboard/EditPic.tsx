
import React, { Dispatch, useEffect, useState } from 'react';
import superagent from 'superagent';
import {
  FormControl, InputLabel, Select, MenuItem,
} from '@material-ui/core';
// import type { Ipicture, PictureContext } from 'src/Providers/PicsProvider';
import type { Ibook } from '../../redux/mapStoreToProps';
import utils from './utils';

async function editPicAPI(body: any, editPic: any, auth: any, dispatch: Dispatch<unknown>, setPicBody: any): Promise<boolean> {
  if (body.title === '') delete body.title;
  if (body.type === '') body.type = 'otherPics';
  if (body.url === '') delete body.url;
  if (body.comments === '') delete body.comments;
  let r;
  try {
    r = await superagent.put(`${process.env.BackendUrl}/book/${editPic._id}`)
      .set('Authorization', `Bearer ${auth.token}`).set('Accept', 'application/json')
      .send(body);
  } catch (e) { console.log((e as Error).message); return false; }
  if (r.status === 200) {
    dispatch({ type: 'EDIT_PIC', picData: {} });
    dispatch({ type: 'SHOW_TABLE', showTable: true });
    setPicBody({
      youthName: '', youthURL: '', type: '', comments: '',
    });
    window.location.reload();
    return Promise.resolve(true);
  }
  return Promise.resolve(false);
}

function handleCancel(dispatch: Dispatch<unknown>, setPicBody: (arg0: any) => void) {
  dispatch({ type: 'EDIT_PIC', picData: {} });
  dispatch({ type: 'SHOW_TABLE', showTable: true });
  setPicBody({
    title: '', url: '', type: '', comments: '',
  });
}

interface IpicButtonProps {
  editPic: any, setPicBody: (arg0: any) => void, picBody: any, auth: any, dispatch: Dispatch<unknown>
}
const PicButton = (props: IpicButtonProps) => {
  const {
    editPic, setPicBody, picBody, auth, dispatch,
  } = props;
  return (
    <div style={{ marginLeft: '50%', marginTop: '10px' }}>
      {editPic._id ? (
        <button
          style={{ display: 'relative', marginRight: '20px' }}
          type="button"
          id="cancel-edit-pic"
          onClick={() => handleCancel(dispatch, setPicBody)}
        >
          Cancel
        </button>
      ) : null}
      <button
        style={{ display: 'relative' }}
        type="button"
        id="submitPic"
        onClick={
          editPic._id ? () => editPicAPI(picBody, editPic, auth, dispatch, setPicBody) : () => utils.createBook(picBody, auth)
        }
      >
        {editPic._id ? 'Edit ' : 'Add '}
        Pic
      </button>
    </div>
  );
};

interface ImakeDropdownProps {
  editPic: Ibook,
  picBody: any,
  onChange: (arg0: any) => void,
}
function MakeDropdown(props: ImakeDropdownProps): JSX.Element {
  const { editPic, picBody, onChange } = props;
  const value = picBody.type === '' && editPic.type ? editPic.type : picBody.type;
  return (
    <FormControl fullWidth>
      <InputLabel id="demo-simple-select-label">Category</InputLabel>
      <Select
        labelId="demo-simple-select-label"
        id="demo-simple-select"
        value={value || ''}
        label="Category"
        onChange={onChange}
      >
        <MenuItem value="youthPics">Youth Pics</MenuItem>
        <MenuItem value="familyPics">Family Pics</MenuItem>
        <MenuItem value="musicPics">Music Pics</MenuItem>
        <MenuItem value="habitat">Habitat</MenuItem>
        <MenuItem value="otherPics">Other Pics</MenuItem>
      </Select>
    </FormControl>
  );
}

interface IradioButtonsProps {
  editPic: Ibook, comments: string, onChange: (...args: any) => void
}
const RadioButtons = (props: IradioButtonsProps) => {
  const { comments, onChange, editPic } = props;
  let hideCaption = true;
  if (comments === 'showCaption' || (comments === '' && editPic.comments === 'showCaption')) hideCaption = false;
  return (
    <div>
      <label htmlFor="hide-caption" style={{ position: 'relative', display: 'inline-block', width: '130px' }}>
        <input
          id="hide-caption"
          type="radio"
          name="hide-caption"
          value="hideCaption"
          checked={hideCaption}
          onChange={onChange}
          className="form-check-input"
          style={{ minWidth: 0 }}
        />
        Hide Caption
      </label>
      <label htmlFor="show-caption" style={{ position: 'relative', display: 'inline-block', width: '130px' }}>
        <input
          type="radio"
          name="show-caption"
          value="showCaption"
          checked={!hideCaption}
          onChange={onChange}
          className="form-check-input"
          style={{ minWidth: 0 }}
        />
        Show Caption
      </label>
    </div>
  );
};

interface IeditPicProps {
  editPic: Ibook,
  auth: any, dispatch: Dispatch<unknown>
}
export function EditPic(props: IeditPicProps): JSX.Element {
  const { editPic, auth, dispatch } = props;
  const [picBody, setPicBody] = useState({
    title: '', url: '', type: '', comments: '',
  });
  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => handleCancel(dispatch, setPicBody), []);
  return (
    <div
      className="material-content elevation3"
      style={{ maxWidth: '320px', margin: '30px auto' }}
    >
      <h4 className="material-header-h4">
        {editPic._id ? 'Edit ' : 'Add '}
        Pictures
      </h4>
      <form id="picsForm">
        <label htmlFor="Title">
          Picture Title
          <input
            id="Title"
            placeholder={editPic.title}
            value={picBody.title}
            onChange={(evt) => setPicBody({ ...picBody, title: evt.target.value })}
          />
        </label>
        <label htmlFor="URL">
          Image Address
          <input
            style={{ marginBottom: '20px' }}
            id="URL"
            placeholder={editPic.url}
            value={picBody.url}
            onChange={(evt) => setPicBody({ ...picBody, url: evt.target.value })}
          />
        </label>
        <MakeDropdown editPic={editPic} picBody={picBody} onChange={(evt:any) => setPicBody({ ...picBody, type: evt.target.value })} />
        {/* {Forms.makeDropdown('type', 'Category', categoryValue, (evt) => setPicBody({ ...picBody, type: evt.target.value }), options)} */}
        <RadioButtons editPic={editPic} comments={picBody.comments} onChange={(evt: any) => setPicBody({ ...picBody, comments: evt.target.value })} />
        <PicButton editPic={editPic} setPicBody={setPicBody} picBody={picBody} auth={auth} dispatch={dispatch} />
      </form>
    </div>
  );
}

