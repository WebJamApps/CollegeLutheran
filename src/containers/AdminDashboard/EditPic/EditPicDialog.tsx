import { useContext } from 'react';
import { AuthContext } from 'src/providers/Auth.provider';
import { ContentContext } from 'src/providers/Content.provider';
import {
  Box,
  Button,
  Checkbox, Dialog, DialogActions, DialogContent, DialogTitle,
  FormControl, FormControlLabel, FormGroup, InputLabel, MenuItem, Select, SelectChangeEvent,
} from '@mui/material';
import libUtils from 'src/lib/commonUtils';
import { defaultPic } from '../utils';
import { EditPicTextField } from './EditPicTextField';
import picUtils from '../pictures.utils';

function checkDisabled(editPic: typeof defaultPic):boolean {
  return !!(editPic.title && editPic.url);
}

interface IeditPicDialogProps {
  onClose:() => void,
  editPic: typeof defaultPic, setEditPic: (arg0:typeof defaultPic) => void,
}
export function EditPicDialog({ editPic, setEditPic, onClose }: IeditPicDialogProps) {
  const { auth } = useContext(AuthContext);
  const { getPictures } = useContext(ContentContext);
  const showHideCaption = libUtils.makeShowHideChecked(setEditPic, editPic, 'showCaption');
  const handleChange = (event: SelectChangeEvent) => {
    const { target: { value } } = event;
    setEditPic({ ...editPic, type: event.target.value });
    return value;
  };
  // const isFormValid = () => editPic.title !== '' && editPic.url !== '';
  return (
    <Dialog
      disableEnforceFocus
      disableAutoFocus
      className="editPicDialog"
      open={!!editPic._id}
      onClose={() => setEditPic(defaultPic)}
    >
      <DialogTitle>Edit Picture</DialogTitle>
      <DialogContent sx={{ padding: '10px 10px' }}>
        <EditPicTextField
          value={editPic.url}
          label="* URL"
          onChange={(evt) => {
            const { target: { value } } = evt;
            setEditPic({ ...editPic, url: value });
            return value;
          }}
        />
        <EditPicTextField
          value={editPic.title}
          label="* Title"
          onChange={(evt) => {
            const { target: { value } } = evt;
            setEditPic({ ...editPic, title: value });
            return value;
          }}
        />
        <Box sx={{ minWidth: 120, marginTop: '20px' }}>
          <FormControl fullWidth>
            <InputLabel id="demo-simple-select-label">Type</InputLabel>
            <Select
              labelId="demo-simple-select-label"
              id="demo-simple-select"
              value={editPic.type}
              label="Type"
              onChange={handleChange}
            >
              <MenuItem value="musicPics">musicPics</MenuItem>
              <MenuItem value="familyPics">familyPics</MenuItem>
              <MenuItem value="youthPics">youthPics</MenuItem>
              <MenuItem value="habitatPics">habitatPics</MenuItem>
              <MenuItem value="otherPics">otherPics</MenuItem>
            </Select>
          </FormControl>
        </Box>
        <FormGroup>
          <FormControlLabel
            control={(
              <Checkbox
                checked={editPic.comments === 'showCaption'}
                onClick={showHideCaption}
              />
          )}
            label="Show Title In Caption"
          />
        </FormGroup>
      </DialogContent>
      <DialogActions>
        <>
          <Button
            disabled={!checkDisabled(editPic)}
            size="small"
            variant="contained"
            className="updatePicButton"
            // disabled={!isFormValid()}
            onClick={() => {
              (async () => {
                await picUtils.updatePic(editPic, auth, getPictures, setEditPic, onClose);
              })();
            }}
          >
            Update
          </Button>
          <Button
            style={{ color: 'red' }}
            size="small"
            className="deletePicButton"
            // eslint-disable-next-line @typescript-eslint/no-floating-promises
            onClick={() => { picUtils.deletePic(editPic, getPictures, auth, setEditPic, onClose); }}
          >
            Delete
          </Button>
          <Button
            size="small"
            className="cancelPicButton"
            onClick={() => { setEditPic(defaultPic); }}
          >
            Cancel
          </Button>
        </>
      </DialogActions>
    </Dialog>
  );
}

