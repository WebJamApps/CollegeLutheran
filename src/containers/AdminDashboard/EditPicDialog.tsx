import { useContext } from 'react';
import { AuthContext } from 'src/providers/Auth.provider';
import { ContentContext } from 'src/providers/Content.provider';
import {
  Button,
  Checkbox, Dialog, DialogActions, DialogContent, DialogTitle, FormControlLabel, FormGroup,
} from '@mui/material';
import { defaultPic } from './utils';
import { EditPicTextField } from './EditPicTextField';
import picUtils from './pictures.utils';
import libUtils from 'src/lib/commonUtils';
function checkDisabled(editPic: typeof defaultPic):boolean {
  return !!(editPic.title && editPic.url);
}

interface IeditPicDialogProps {
  onClose:() => void,
  editPic: typeof defaultPic, setEditPic: (arg0:typeof defaultPic) => void,
}
export function EditPicDialog({ editPic, setEditPic, onClose }: IeditPicDialogProps) {
  const { auth } = useContext(AuthContext);
  const { getPictures } = useContext(ContentContext);
  const showHideCaption = libUtils.makeShowHideChecked(setEditPic, editPic, 'showCaption');
  return (
    <Dialog
      disableEnforceFocus
      disableAutoFocus
      className="editPicDialog"
      open={!editPic._id ? false : true}
      onClose={() => setEditPic(defaultPic)}
    >
      <DialogTitle>Edit Picture</DialogTitle>
      <DialogContent sx={{ padding: '10px 10px' }}>
        <EditPicTextField
          value={editPic.url}
          label="* URL"
          onChange={(evt) => {
            const { target: { value } } = evt;
            setEditPic({ ...editPic, url: value });
            return value;
          }}
        />
        <EditPicTextField
          value={editPic.title}
          label="* Title"
          onChange={(evt) => {
            const { target: { value } } = evt;
            setEditPic({ ...editPic, title: value });
            return value;
          }}
        />
        <FormGroup>
          <FormControlLabel
            control={(
              <Checkbox
                checked={editPic.comments === 'showCaption'}
                onClick={showHideCaption}
              />
          )}
            label="Show Title In Caption"
          />
        </FormGroup>
      </DialogContent>
      <DialogActions>
        <>
          <Button
            disabled={!checkDisabled(editPic)}
            size="small"
            variant="contained"
            className="updatePicButton"
            onClick={() => {
              (async () => {
                await picUtils.updatePic(editPic, auth, getPictures, setEditPic, onClose);
              })();
            }}
          >
            Update
          </Button>
          <Button
            style={{ color: 'red' }}
            size="small"
            className="deletePicButton"
            // eslint-disable-next-line @typescript-eslint/no-floating-promises
            onClick={() => { picUtils.deletePic(editPic, getPictures, auth, setEditPic, onClose); }}
          >
            Delete
          </Button>
          <Button
            size="small"
            className="cancelPicButton"
            onClick={() => { setEditPic(defaultPic); }}
          >
            Cancel
          </Button>
        </>
      </DialogActions>
    </Dialog>
  );
}

