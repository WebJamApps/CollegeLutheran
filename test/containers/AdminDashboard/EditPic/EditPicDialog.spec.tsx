import { EditPicDialog } from 'src/containers/AdminDashboard/EditPic/EditPicDialog';
import renderer from 'react-test-renderer';
import { defaultPic } from 'src/containers/AdminDashboard/utils';

describe('EditPicDialog', () => {
  it('renders EditPicDialog', () => {
    const props = {
      onClose: jest.fn(), editPic: defaultPic, setEditPic: jest.fn(),
    };
    const epd = renderer.create(<EditPicDialog {...props} />).toJSON();
    expect(epd).toMatchSnapshot();
  });
  it('handles events for EditPicDialog for url field', () => {
    const props = {
      onClose: jest.fn(), editPic: defaultPic, setEditPic: jest.fn(),
    };
    const evt = { target: { value: 'url' } };
    const result = renderer.create(<EditPicDialog {...props} />).root;
    expect(result.findByProps({ label: '* URL' }).props.onChange(evt)).toBe('url');
  });
  it('handles events for EditPicDialog for title field', () => {
    const props = {
      onClose: jest.fn(), editPic: defaultPic, setEditPic: jest.fn(),
    };
    const evt = { target: { value: 'title' } };
    const result = renderer.create(<EditPicDialog {...props} />).root;
    expect(result.findByProps({ label: '* Title' }).props.onChange(evt)).toBe('title');
  });
  it('handles onClose for EditPicDialog', () => {
    const props = {
      editPic: defaultPic, onClose: jest.fn(), setEditPic: jest.fn(),
    };
    const result = renderer.create(<EditPicDialog {...props} />).root;
    result.findByProps({ className: 'editPicDialog' }).props.onClose();
    expect(props.setEditPic).toHaveBeenCalled();
  });
  // it('handles onClick for EditPicDialog', () => {
  //   picUtils.updatePic = jest.fn();
  //   picUtils.deletePic = jest.fn();
  //   const props = {
  //     editPic: defaultPic, onClose: jest.fn(), setEditPic: jest.fn(),
  //   };
  //   const result = renderer.create(<EditPicDialog {...props} />).root;
  //   result.findByProps({ className: 'updatePicButton' }).props.onClick();
  //   expect(picUtils.updatePic).toHaveBeenCalled();
  // });
});
