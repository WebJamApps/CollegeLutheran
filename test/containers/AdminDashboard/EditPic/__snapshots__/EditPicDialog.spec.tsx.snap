// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPicDialog renders EditPicDialog 1`] = `
<div
  className="editPicDialog"
  disableAutoFocus={true}
  disableEnforceFocus={true}
  onClose={[Function]}
  open={false}
>
  <div>
    Edit Picture
  </div>
  <div
    sx={
      {
        "padding": "10px 10px",
      }
    }
  >
    <input
      fullWidth={true}
      label="* URL"
      onChange={[Function]}
      sx={
        {
          "marginTop": "20px",
        }
      }
      type="text"
      value=""
    />
    <input
      fullWidth={true}
      label="* Title"
      onChange={[Function]}
      sx={
        {
          "marginTop": "20px",
        }
      }
      type="text"
      value=""
    />
    <div>
      <div
        control={
          <Checkbox
            checked={false}
            onClick={[Function]}
          />
        }
        label="Show Title In Caption"
      />
    </div>
  </div>
  <div>
    <button
      className="updatePicButton"
      disabled={true}
      onClick={[Function]}
      size="small"
      variant="contained"
    >
      Update
    </button>
    <button
      className="deletePicButton"
      onClick={[Function]}
      size="small"
      style={
        {
          "color": "red",
        }
      }
    >
      Delete
    </button>
    <button
      className="cancelPicButton"
      onClick={[Function]}
      size="small"
    >
      Cancel
    </button>
  </div>
</div>
`;
